src/render/sdlgpu/SDL_shaders_gpu.c

```diff
@@ -0,0 +1,250 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+#if SDL_VIDEO_RENDER_GPU
+
+#include "SDL_shaders_gpu.h"
+
+/* SDL_Gpu shader implementation */
+
+typedef struct GPU_ShaderModuleSource
+{
+    const unsigned char *code;
+    unsigned int code_len;
+    SDL_GpuShaderFormat format;
+} GPU_ShaderModuleSource;
+
+// FIXME Please fix this in the build system!
```

Comment:
These will have to be `#ifdef` rather than just `#if`, since that's how cmakedefines come out https://github.com/thatcosmonaut/SDL/blob/gpu/include/build_config/SDL_build_config.h.cmake#L447

Reply:
Looks like SDL defines a bunch of undefined macros as 0s here, at least for the render backends: https://github.com/libsdl-org/SDL/blob/a7bed810b3499d157655ae6dcc7cd5e8a34fa549/src/SDL_internal.h#L194

GPU also uses `#if` checks on those, which is technically not correct (clang throws a warning).

Maybe we could do something similar for GPU backends. I like them being 1/0 macros because it makes the implementation of `GPU_FillSupportedShaderFormats` neat, but it's not a big deal.

Reply:
Oh, good catch - we can do the same for GPU.